main/src/library/Concurrent/Channel/Timer.flix

```diff
@@ -79,5 +79,10 @@ namespace Concurrent/Channel/Timer {
         import static java.lang.Thread.sleep(Int64, Int32): Unit & Impure;
         let millis = ns  /  1_000_000i64;
         let nanos  = ns mod 1_000_000i64;
-        sleep(millis, Int64.clampToInt32(nanos, 0i32, 1_000_000i32)); c <- (); ()
+        if (millis < 0i64 or nanos < 0i64) {
```

Comment:
I couldn't seem to figure out if the `nanos < 0i64` is redundant or not.

Reply:
I had to look it up too but `rem` has the sign of its lefhand side and `mod` is always positive. In either case `nanos` is positive if `millis` is positive if `ns` is positive. Maybe its the easiest to see if `s < 0i64` at the start and then have the nanos, millis computations in the positive case.

Reply:
Good idea :-)